infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/project/OpenShiftWorkspaceServiceAccount.java

```diff
@@ -142,6 +151,37 @@ private void createViewRole(OpenShiftClient osClient, String name) {
     osClient.roles().inNamespace(projectName).create(viewRole);
   }
 
+  private void createStopWorkspacesRole(OpenShiftClient osClient, String name) {
```

Comment:
I would probably move the logic of granting the rights on the level above to the `OpenShiftProjectFactory` / `getOrCreate` + extract the logic of the role / rb to a separate class smth like `OpenShiftStopWorkspaceRoleProvisioner`   with docs and tests